src/coreComponents/mesh/PerforationData.cpp

```diff
@@ -126,8 +126,12 @@ void PerforationData::computeWellTransmissibility( MeshLevel const & mesh,
 
     // if the well transmissibility has been read from the XML
     // then skip the computation of the well transmissibility carried out below
-    if( m_wellTransmissibility[iperf] > 0 )
+    if( m_wellTransmissibility[iperf] >= 0 )
     {
+      GEOSX_LOG_RANK_IF( isZero( m_wellTransmissibility[iperf] ),
+                         "\n \nWarning! A perforation is defined with a zero transmissibility! \n"
```

Comment:
Is there a way to tell which one?
If not, it's OK.

Reply:
Yes, I modified this error message to show which well has the zero transmissibility. I also updated the error messages in various places in the well construction code.